plugins/ltac/tacinterp.ml

```diff
@@ -2017,6 +2017,30 @@ end
 (* globalization + interpretation *)
 
 
+let warning_fragile_intros =
+  CWarnings.create ~default:CWarnings.Disabled ~name:"fragile-intros" ~category:CWarnings.CoreCategories.pedantic
+    (fun () -> strbrk "intros is fragile, specify the names." ++ fnl ())
+
+let intro_patterns_deprecated b l loc : unit Proofview.tactic =
+  Proofview.Goal.enter begin fun gl ->
+    let old_hyps = Proofview.Goal.hyps gl in
+    Tactics.intro_patterns b l >>=
+    fun () -> Proofview.Goal.enter begin fun gl ->
+      let new_hyps = Proofview.Goal.hyps gl in
+      let old_ids = List.map Context.Named.Declaration.get_id old_hyps in
+      let new_ids = List.map Context.Named.Declaration.get_id new_hyps in
+      let nnew = List.length new_ids in
+      let nold = List.length old_ids in
+      let names, rest = CList.chop (nnew - nold) new_ids in
+      let names = List.rev names in
+      if names <> [] && Option.has_some loc then
+        warning_fragile_intros ?loc
+          ~quickfix:[Feedback.Quickfix.make ~loc:Option.(get loc) Pp.(str "intros " ++ prlist_with_sep spc Id.print names)] ();
```

Comment:
Will this behave acceptably if I have `split; intros` where the executions of `intros` produce different names?

Reply:
>In the code below I think I call this only if the toplevel tactic is intros.

I don't think that's correct.

If you have something that uses the quickfix you can test with
~~~coq
Goal (forall x:nat, x = x) /\ (forall y:bool, y = y).
  split; intros.
~~~

Reply:
also worth considering
~~~coq
Ltac the_tac := split; intros; reflexivity.

Goal (forall x:nat, x = x) /\ (forall y:bool, y = y).
  split; intros.
~~~

Reply:
After testing,
```coq 
Goal (forall x:nat, x = x) /\ (forall y:bool, y = y).
  split; intros.
```
does not seem to trigger a warning

Reply:
>I don't think that's correct.

I was wrong.

Reply:
ok, so I consider this discussion closed